eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php

```diff
@@ -539,4 +541,16 @@ public function loadAllLocations($offset, $limit)
 
         return $this->locationMapper->createLocationsFromRows($rows);
     }
+
+    /**
+     * Returns true if given location is a tree root.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Location $location
+     *
+     * @return bool
+     */
+    public function isRootLocation(Location $location): bool
```

Comment:
Let's not add methods on Persistence which does not involve some logic to interact with storage engine.

just a simple private method on DomainMapper is much better and simpler solution here.